FurCoNZ/Services/Payment/StripePaymentProvider.cs

```diff
@@ -0,0 +1,141 @@
+            _options = options;
+            _orderService = orderService;
+            _logger = logger;
+            _paymentIntentService = new PaymentIntentService();
+        }
+
+        public async Task AddStripeSessionToOrderAsync(int orderId, string sessionId, CancellationToken cancellationToken = default)
+        {
+            var stripeSession = new StripeSession
+            {
+                Id = sessionId,
+                State = StripeSessionState.None,
+                OrderId = orderId,
+            };
+
+            await _dbContext.StripeSessions.AddAsync(stripeSession, cancellationToken);
+            await _dbContext.SaveChangesAsync(cancellationToken);
+        }
+
+        public async Task FufillOrderFromCompletedSessionAsync(Session session, CancellationToken cancellationToken = default)
+        {
+            var stripeSession = await _dbContext.StripeSessions
+                .SingleOrDefaultAsync(s => s.Id == session.Id, cancellationToken);
+
+            if (stripeSession == null)
+            {
+                _logger.LogError($"session does not exist. Has data been lost? ({session.Id})");
+                return;
+            }
+
+
+            if (string.IsNullOrEmpty(session.PaymentIntentId))
+            {
+                _logger.LogCritical($"No Payment Intent associated with Checkout Session ({session.Id})");
+                return;
+            }
+
+            if (session.PaymentIntent == null)
+                session.PaymentIntent = await _paymentIntentService.GetAsync(session.PaymentIntentId, cancellationToken: cancellationToken);
+            
+            // Record the patyment intent for future refunds.
+            stripeSession.PaymentIntent = session.PaymentIntentId;
+
+            if ((stripeSession.State == StripeSessionState.None || stripeSession.State == StripeSessionState.Processing) && 
+                session.PaymentIntent.Status == "succeeded")
+            {
+                var amountReceived = (int)session.PaymentIntent.AmountReceived.Value;
+                // Assume the fee was included in the price during checkout when IncludeFee is enabled.
+                if (_options.Value.IncludeFee)
+                    amountReceived = amountReceived - (amountReceived * 29 / 1000) - 30;
+
+                await _orderService.AddReceivedFundsForOrder(
+                    stripeSession.OrderId, amountReceived,
+                    "stripe", session.PaymentIntentId, 
+                    session.PaymentIntent.Created.Value,
+                    cancellationToken);
+
+                stripeSession.State = StripeSessionState.Completed;
+                // TODO: Record audit log of payment received.
+            }
+
+            await _dbContext.SaveChangesAsync();
+        }
+
+        public async Task ProcessChargeAsync(Charge charge, CancellationToken cancellationToken)
+        {
+            // For now, only process refunds.
+            if(charge.Refunds.Any())
```

Comment:
Small style change: space between keywords
```suggestion
            if (charge.Refunds.Any())
```